### PR TITLE
Add search logging

### PR DESCRIPTION
**Pull request by ammsa**
This is an unfinished pull request.

Adds a logging endpoint that we can use to point logging requests.

User behaviour logging still needs to be done.

### CODE REVIEW

**Comment by KamyarGhajar** [comment]
@ammsa I'll look into it.

**Review by KamyarGhajar — APPROVED** [review]
I have added the logging base + fixed conflicts with master recent updates

**Comment by KamyarGhajar** [comment]
@ammsa Please look into this merge request.

**Comment by ammsa** [comment]
> @ammsa Please look into this merge request.

There's too many changes unrelated to logging in this pull request. Can you make a new pull request with only changes related to logging?

Instead of having the logger.js code in the document-viewer, use the callbacks in the document-viewer

**Comment by KamyarGhajar** [comment]
@ammsa I have finalized the changes as you mentioned. Also I have made use of the event triggers too.

File: docker-compose.yml [hunk 1]
```diff
@@ -47,8 +47,8 @@ services:
       context: ./web/Web
       dockerfile: ./compose/production/postgres/Dockerfile
     volumes:
-      - /postgres_data_local:/var/lib/postgresql/data
-      - /postgres_backup_local:/backups
+      - ./postgres_data_local:/var/lib/postgresql/data
+      - ./postgres_backup_local:/backups
```
Comment (ammsa) [inline docker-compose.yml:51]: Let's not worry about this for this pull request. Having the dot breaks things in my machine and on the tuna server.

File: web/Web/config/logger.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        event = body['event']
+        data = body['data']
+        message = body['message']
+
+        log = {
+            'user': self.request.user.username,
+            'timestamp': timestamp,
+            'event': event,
+            'data': data,
+            'message': message
```
Comment (ammsa) [inline web/Web/config/logger.py:28]: Is message being used for anything? We can probably delete it.

File: web/Web/web/search/templates/search/search.html [hunk 3]
```diff
@@ -51,124 +58,147 @@
 {% include 'modals/document_modal.html' %}
 <!-- Judging criteria modal -->
 {% include 'modals/additional_judging_criteria_modal.html' %}
-<!-- Discovery failed to receive judgment toast -->
-{% include 'toasts/discovery_failed_to_receive_judgment.html' %}
```
Comment (ammsa) [inline web/Web/web/search/templates/search/search.html:55]: Why is this removed?

File: web/Web/web/search/templates/search/search.html [hunk 4]
```diff
@@ -51,124 +58,147 @@
...
-              $(".CALFailedToReceiveJudgmentSpinner").addClass("d-none");
-            },500);
-            $('#toast_CALFailedToReceiveJudgment').toast('show');
-          }
-        });
-
-        function HtmlDecode(s) {
-            var el = document.createElement("div");
-            el.innerHTML = s;
-            return el.innerText || el.textContent;
```
Comment (ammsa) [inline web/Web/web/search/templates/search/search.html:129]: It's much harder to track changes in the code with long unrelated changes. Let's keep it the same for now and have the logging changes added.

File: web/Web/web/static/js/document-viewer.js [hunk 5]
```diff
@@ -1244,9 +1259,14 @@ docView.prototype = {
     return this.triggerEvent("beforeDocumentLoad", [docid]);
   },
 
-  afterDocumentLoad: function(docid) {
+  afterDocumentLoad: function(docid, rel) {
```
Comment (ammsa) [inline web/Web/web/static/js/document-viewer.js:1262]: Let's just call `previouslyJudgedDocs` from here and get the rel value from inside the function. It's better to have just the docid parameter for this function.

**Review by ammsa — COMMENTED** [review]
Much cleaner than before. I wrote some comments and feedback.

File: web/Web/config/logger.py (hunk 2 above)
Comment (KamyarGhajar) [reply web/Web/config/logger.py:28]: I will remove it.

File: web/Web/web/search/templates/search/search.html (hunk 3 above)
Comment (KamyarGhajar) [reply web/Web/web/search/templates/search/search.html:55]: This is not removed actually, the file reformatting has led to these kinds of false removed code alerts. I will fix it so you can check them more easily.

File: web/Web/web/static/js/document-viewer.js (hunk 5 above)
Comment (KamyarGhajar) [reply web/Web/web/static/js/document-viewer.js:1262]: I thought accessing another .js file variable like `previouslyJudgedDocs` would be more problematic and opposes the original purpose of event trigger mechanism you mentioned. I'll arrange to change them back but I need to understand the architecture more thoroughly I guess.
I faced so many formatting issues with PyCharm defaults. The faulty code indents makes the code less readable I'm afraid so someday we might need a refactoring.

**Comment by KamyarGhajar** [comment]
@ammsa I guess now it is what you desired my friend, please check.

File: web/Web/web/static/js/document-viewer.js [hunk 6]
```diff
@@ -1259,14 +1257,14 @@ docView.prototype = {
     return this.triggerEvent("beforeDocumentLoad", [docid]);
   },
 
-  afterDocumentLoad: function(docid, rel) {
+  afterDocumentLoad: function(docid) {
     "use strict";
-    return this.triggerEvent("afterDocumentLoad", [docid, rel]);
```
Comment (ammsa) [inline web/Web/web/static/js/document-viewer.js]: You can return the rel value here, but extract it in the function itself rather than passing it as a parameter. That's what I meant earlier.

Comment (KamyarGhajar) [reply web/Web/web/static/js/document-viewer.js]: Done!

**Comment by KamyarGhajar** [comment]
I guess it is ready to be merged now @ammsa , isn't it?

**Comment by ammsa** [comment]
> I guess it is ready to be merged now @ammsa , isn't it?

I didn't have time to test it yet but it seems like it's ready.

**Comment by ammsa** [comment]
I'll test it sometime soon and merge it after.
